### PR TITLE
[test] Add chart install and upgrade testing

### PR DESCRIPTION
**Pull request by jkoelker**
* Adds a testing script that tests the charts using `ct lint-and-install --upgrade`.
* Backports a fix from upstream `dex` chart to allow install and upgrade to succeed (https://github.com/helm/charts/pull/18380)
* Clean up EOL whitespace

### CODE REVIEW

File: test/ct-e2e.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+debug: true
+target-branch: dev
+chart-dirs:
+  - stable
+  - staging
+excluded-charts:
+  - common
+  - dex-controller # Moved to a different helm repo
+  - azuredisk-csi-driver  # DCOS-62804
```
Comment (jkoelker) [inline test/ct-e2e.yaml]: Its my hope that once all the tickets are closed we can remove the special *e2e* variant of the config and both linting and install and upgrade tests can share just the one config.

File: stable/dex/templates/config-openssl.yaml [hunk 2]
```diff
@@ -8,10 +8,6 @@ metadata:
     heritage: "{{ .Release.Service }}"
     release: "{{ .Release.Name }}"
   name: {{ template "dex.fullname" . }}-openssl-config
-  annotations:
```
Comment (alejandroEsc) [inline stable/dex/templates/config-openssl.yaml]: Why these changes to dex? Could you provide some context, just want to know why we are no longer using this hook, its a fairly large change so want to be careful.

Comment (jkoelker) [reply stable/dex/templates/config-openssl.yaml]: The context is in the commit message for this commit. Basically this causes the configmap to be be deleted immediatly after creation, and the `grpc-tls` and `web-tls` jobs which depend on it fail, causing the install to fail. This is the backport from https://github.com/helm/charts/pull/18380, but since the charts have diverged enough I couldn't just `cherry-pick` it ;(.

File: staging/cert-manager-setup/Chart.yaml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 name: cert-manager-setup
 home: https://github.com/mesosphere/charts
-version: 0.1.6
+version: 0.1.7
```
Comment (jkoelker) [inline staging/cert-manager-setup/Chart.yaml]: Technically not really a change to the chart, but `ct` lint detects the new `ci/test-values.yaml` that the e2e test script uses as a change, so...
